File: php/features.php
<?php
declare(strict_types=1);

// WebsiteAnalysisApis SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class WebsiteAnalysisApisFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new WebsiteAnalysisApisBaseFeature();
            case "test":
                return new WebsiteAnalysisApisTestFeature();
            default:
                return new WebsiteAnalysisApisBaseFeature();
        }
    }
}

File: py/features.py
# WebsiteAnalysisApis SDK feature factory

from feature.base_feature import WebsiteAnalysisApisBaseFeature
from feature.test_feature import WebsiteAnalysisApisTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WebsiteAnalysisApisBaseFeature(),
        "test": lambda: WebsiteAnalysisApisTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
